test/unit/config/config-formatters.test.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const sinon = require('sinon')
+
+const formatters = require('../../../lib/config/formatters')
+tap.test('config formatters', (t) => {
+  t.autoend()
+
+  tap.test('array', (t) => {
+    t.autoend()
+
+    t.test('should trim string into array', (t) => {
+      const val = 'opt1, opt2  ,   opt3 , opt4'
+      const options = formatters.array(val)
+      t.same(options, ['opt1', 'opt2', 'opt3', 'opt4'])
+      t.end()
+    })
+
+    t.test('should create an array with 1 element if no comman exists', (t) => {
+      t.same(formatters.array('hello'), ['hello'])
+      t.end()
+    })
+  })
+
+  tap.test('int', (t) => {
+    t.autoend()
+
+    t.test('should parse number string as int', (t) => {
+      t.equal(formatters.int('100'), 100)
+      t.end()
+    })
+
+    t.test('should return isNaN is string is not a number', (t) => {
+      t.ok(isNaN(formatters.int('hello')))
+      t.end()
+    })
+
+    t.test('should parse float as int', (t) => {
+      const values = ['1.01', 1.01]
+      values.forEach((val) => {
+        t.equal(formatters.int(val), 1)
+      })
+      t.end()
+    })
+  })
+
+  tap.test('float', (t) => {
+    t.autoend()
+
+    t.test('should parse number string as float', (t) => {
+      t.equal(formatters.float('100'), 100)
+      t.end()
+    })
+
+    t.test('should return isNaN is string is not a number', (t) => {
+      t.ok(isNaN(formatters.float('hello')))
+      t.end()
+    })
+
+    t.test('should parse float accordingly', (t) => {
+      const values = ['1.01', 1.01]
+      values.forEach((val) => {
+        t.equal(formatters.float(val), 1.01)
+      })
+      t.end()
+    })
+  })
+
+  tap.test('boolean', (t) => {
+    t.autoend()
+
+    const falseyValues = [null, 'false', 'f', 'no', 'n', 'disabled', '0']
+    falseyValues.forEach((val) => {
+      t.test(`should map ${val} to false`, (t) => {
+        t.equal(formatters.boolean(val), false)
+        t.end()
+      })
+    })
+
+    // these are new tests but do not want to change behavior of this formatter
+    // but anything that is not a falsey value above is true ¯\_(ツ)_/¯
```

Comment:
/shrug agreed, but this is insane logic that we support so many things that aren't `false` as false, and everything else is true. probably worth coming back to when we have something else that requires a breaking change.

Reply:
yea or i can just be snarky about it until the end of time

Reply:
Is this SDD (snark-driven development)?